### ISSUE TITLE
Earth disappears when changing focus

### ISSUE THREAD

**Issue by carteremmart** [OpenSpace/OpenSpace#1455]
When changing focus to a distant object, the earth disappears before its offscreen, leaving a transparent atmosphere only.

**Comment by Roxeena** [comment]
Related to #1121?

**Comment by GPayne** [comment]
Are there any additional steps to reproduce this? I have seen this happen occasionally, but am currently unable to reproduce this.

**Comment by carteremmart** [comment]
For me it is happening consistently.  I will go back over it and see if I can document a range where is does / does not happen.
________________________________

**Comment by estrelanova** [comment]
With me it happens regularly, but not with all objects in the same manner. If I focus on Mars, all layers dissapear. If I focus on the Moon, Earth stays "intact", if I focus on the Sun, only parts of Earth's layers dissapear. If I focus on Jupiter or Saturn, not even a sphere stays visible, just the trails of the Planet and of the ISS.

**Comment by WeirdRubberDuck** [comment]
Yes, it seems like the behavior varies depending on the distance to the newly targeted object. The further away the target is, the more of the Earth rendering disappears

**Comment by alexanderbock** [comment]
> 
> 
> Related to #1121?

Yes, I think it is related

**Comment by alexanderbock** [comment]
Worked in 0.15.2

**Comment by alexanderbock** [comment]
Trying milkyway and disabling adaptive stereo by default?

**Comment by alexanderbock** [comment]
Describing the problem a bit more. It's all related to the adaptive stereo setting that causes the entire scenegraph to be shrunk down depending on the distance from the camera to the focus node.  With our default settings, when switching the target from the Earth to Mars, the world gets scaled up so much that we are getting precision errors (i think in the position g-buffer) which cause the Earth to disappear.
Two potential options to solve this:

Disabling the adaptive stereo
This works for preventing the Earth from disappearing as we are always working with values that work pretty well.  However, it a) destroys the stereo separation in stereo rendering mode so it won't be a solution for that case.  It also kills the galaxy volume rendering as we are running in to floating point overflow problems when rendering the *inside* of the entry-exit box which causes an entire face to disappear, thus causing issues with the milkyway rendering

Changing the way adaptive stereo works
Right now, we are only specifying a single "proper" distance mode, but maybe we can come up with something smarter than that. Maybe some nonlinear scaling that would give us the necessary precision on both ends (changing targets in solar system and milkyway)


For the time being, and realizing that only the minority of people use stereo rendering mode, I changed the default of the Distance-to-depth-of-focus setting that controls the adaptive stereo to some insane value (200000m, compared to the previous 8m).  This will cause all stereo rendering to be flat (thus needing to change the property to bring out 3D again), but in exchange we are getting the expected behavior of planets not disappearing and the galaxy rendering

**Comment by alexanderbock** [comment]
Different value, maybe 30000?  *yikes*

**Comment by alexanderbock** [comment]
Trying to find and upper and a lower bound that is acceptable for everything:


200000:  Problems with Earth atmosphere

**Comment by micahnyc** [comment]
14k working good - no earth disappear, no iss/atmosphere issue. Milkyway volume working.

**Comment by Roxeena** [comment]
A value of around 3000 (or below) works good with the JWST pointing towards the Orion Nebula. If the value is higher the model becomes black, see video

https://user-images.githubusercontent.com/16215103/149802504-ef173e04-9a48-47e5-9dba-29c33a80a58f.mp4

**Comment by alexanderbock** [comment]
Removing this from the milestone as a good implementation of this requires #1738 and anything else would be hacked (as visible in the messages above trying to find a good value that matches nothing perfectly)
